### PR TITLE
Add Login functionality

### PR DESCRIPTION
**Pull request by chamika**
Add Login functionality to be shown at the first run of the application. Login will request user to enter a username and select a country

### CODE REVIEW

File: PCSA/PCSA/LoginViewController.swift [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+    var locationName = Constants.CONTACT_LOCATIONS[0]
+
+    // MARK: Properties
+    
+    @IBOutlet weak var locationPicker: UIPickerView!
+    @IBOutlet weak var userNameText: UITextField!
+    
+    @IBAction func loginButtonTap(sender: AnyObject) {
+        
+        let username = userNameText.text
```
Comment (shruti-gupta) [inline PCSA/PCSA/LoginViewController.swift]: how about using: 

if let username = userNameText.text {

 if username.isEmpty {
UIUtil.showAlert(self, title: "Name", message: "User name is not valid", actions: nil)
 +            return
}

}

File: PCSA/PCSA/LoginViewController.swift [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+    
+    override func viewDidLoad() {
+        super.viewDidLoad()
+
+        self.locationPicker.dataSource = self
+        self.locationPicker.delegate = self
+        self.userNameText.delegate = self
+    }
+
+    override func didReceiveMemoryWarning() {
```
Comment (shruti-gupta) [inline PCSA/PCSA/LoginViewController.swift]: remove methods where you do not change anything. Do it throughout the PR.

File: PCSA/PCSA/LoginViewController.swift [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+    
+    func pickerView(pickerView: UIPickerView, titleForRow row: Int, forComponent component: Int) -> String? {
+        return pickerDataSource[row]
+    }
+    
+    func pickerView(pickerView: UIPickerView, didSelectRow row: Int, inComponent component: Int) {
+        locationName = pickerDataSource[row]
+    }
+    
+    // UITextField Delegate
```
Comment (shruti-gupta) [inline PCSA/PCSA/LoginViewController.swift]: separate picker and textfield extensions.

File: PCSA/PCSA/SpashViewController.swift [hunk 4]
```diff
@@ -10,16 +10,33 @@
...
-
+    
     override func viewDidLoad() {
         super.viewDidLoad()
         
-        performSelector(#selector(SplashViewController.showMainController), withObject: nil, afterDelay: 0)
+        let defaults = NSUserDefaults.standardUserDefaults()
+        let logged = defaults.boolForKey(Configs.KEY_LOGGED)
+        
+        if(logged){
```
Comment (shruti-gupta) [inline PCSA/PCSA/SpashViewController.swift]: correct formatting. Update throughout PR

File: PCSA/PCSA/SpashViewController.swift [hunk 5]
```diff
@@ -10,16 +10,33 @@
 import UIKit
 
 class SplashViewController: UIViewController {
-
+    
     override func viewDidLoad() {
         super.viewDidLoad()
         
-        performSelector(#selector(SplashViewController.showMainController), withObject: nil, afterDelay: 0)
+        let defaults = NSUserDefaults.standardUserDefaults()
```
Comment (shruti-gupta) [inline PCSA/PCSA/SpashViewController.swift]: write logic in comment - what is exactly happening here.

File: PCSA/PCSA/SpashViewController.swift [hunk 6]
```diff
@@ -10,16 +10,33 @@
...
     func showMainController(){
         performSegueWithIdentifier("MainNav", sender: self)
     }
-
+    
+    func showLoginController(){
+        performSegueWithIdentifier("LoginNav", sender: self)
+    }
+    
+    override func didReceiveMemoryWarning() {
```
Comment (shruti-gupta) [inline PCSA/PCSA/SpashViewController.swift]: remove unchanged overridden methods.

File: PCSA/PCSA/LoginViewController.swift (hunk 1 above)
Comment (chamika) [inline PCSA/PCSA/LoginViewController.swift]: the username variable is used after this line(31). Therefore variable's scope should be out of the if condition. Is there any other way to do that?

File: PCSA/PCSA/LoginViewController.swift (hunk 2 above)
Comment (chamika) [inline PCSA/PCSA/LoginViewController.swift]: Done

File: PCSA/PCSA/LoginViewController.swift (hunk 3 above)
Comment (chamika) [inline PCSA/PCSA/LoginViewController.swift]: Didn't get the idea. Please can you describe more?

File: PCSA/PCSA/SpashViewController.swift (hunk 4 above)
Comment (chamika) [inline PCSA/PCSA/SpashViewController.swift]: Done

File: PCSA/PCSA/SpashViewController.swift (hunk 5 above)
Comment (chamika) [inline PCSA/PCSA/SpashViewController.swift]: Added

File: PCSA/PCSA/SpashViewController.swift (hunk 6 above)
Comment (chamika) [inline PCSA/PCSA/SpashViewController.swift]: Removed

**Pull request merged**